### ISSUE TITLE
metamorphic: certain configurations result in test timeouts

### ISSUE THREAD

**Issue by nicktrav** [cockroachdb/pebble#1514]
Prior to #1470, each run of `TestMeta` would complete relatively quickly (within a minute or two). This was well within the grace period given to the nightly metamorphic test harness that runs `stress` on `TestMeta` for three hours (we give the test runner 11,000 seconds == 183.3 mins in which to complete, which allows for some overrun).

After #1470 there are certain configurations that take _much longer_ to complete a single run. These configurations appear to be  "worst case scenarios" for #1470 - small memtable size, small flush split size, high L0 compaction and stop write thresholds.

Consider the following diff to the base configuration (adapted from [this](https://teamcity.cockroachdb.com/viewLog.html?buildId=4377971&buildTypeId=Cockroach_Nightlies_Pebble_Metamorphic&tab=buildLog) nightly run timeout), which seems to highlight the problem:

```diff
--- ./internal/metamorphic/_meta/220210-070230.298/standard-000/OPTIONS.bak	2022-02-15 09:29:50.474474925 -0800
+++ ./internal/metamorphic/_meta/220210-070230.298/standard-000/OPTIONS	2022-02-15 13:37:59.346607150 -0800
@@ -9,16 +9,16 @@
   comparer=pebble.internal.testkeys
   delete_range_flush_delay=0s
   disable_wal=false
-  flush_split_bytes=4194304
+  flush_split_bytes=16
   format_major_version=5
   l0_compaction_concurrency=10
-  l0_compaction_threshold=4
-  l0_stop_writes_threshold=12
+  l0_compaction_threshold=71
+  l0_stop_writes_threshold=71
   lbase_max_bytes=67108864
   max_concurrent_compactions=1
   max_manifest_file_size=134217728
   max_open_files=1000
-  mem_table_size=4194304
+  mem_table_size=1024
   mem_table_stop_writes_threshold=2
   min_compaction_rate=4194304
   min_deletion_rate=0
```

Running this configuration now takes ~133 seconds. Prior to the change, it takes ~7 seconds to complete.

The above is just an attempt to produce a minimal configuration that demonstrates the problem. An example of a configuration that took ~1 hour to complete  can be seen [here](https://gist.github.com/nicktrav/b75231908cfec9390c4dd3c8b7dddcab).

**Comment by jbowens** [comment]
A few thoughts:
* I think `use_disk=true` also exacerbates these cases, forcing the flush/compaction to perform IO for each new file. I think it might be worth disabling disk in all random runs, and relying on the one `use_disk=true` standard configuration for disk coverage. The faster in-memory random runs seem like they'd increase our test coverage more.
* We generate arbitrarily wide range deletes and range keys. I think this problem would be mitigated some by using a weighted distribution for the width of these operations' key ranges. Smaller ranges result in less grandparent fragmentation.
* I suspect we get more testing coverage out of these unusual, slow configurations than we do out of having _many_ operations in a single run. The unusual configurations are great at exercising subtle edge cases around compaction boundaries. I think it might be worthwhile to drop the op count per-run.

**Comment by nicktrav** [comment]
> I think use_disk=true also exacerbates these cases

Agree. Interestingly, this particular configuration was using a mem fs.

> relying on the one use_disk=true standard configuration for disk coverage

Sounds reasonable. This is a quick and easy change to make.

> I think it might be worthwhile to drop the op count per-run.

I assume you mean lower the op count, rather than dropping it entirely? Agree that's probably also a good change to make, as it will let us test out more configurations too.
